pkg/auth/handlers_test.go

```diff
@@ -53,3 +54,21 @@ func TestGetHttpRequestCookieToMetadataHandler(t *testing.T) {
 	req.AddCookie(&jwtCookie)
 	assert.Equal(t, "Bearer a.b.c", handler(ctx, req)["authorization"][0])
 }
+
+func TestGetMetadataEndpointRedirectHandler(t *testing.T) {
+	ctx := context.Background()
+	baseURL, err := url.Parse("http://www.google.com")
+	assert.NoError(t, err)
+	metadataPath, err := url.Parse(MetadataEndpoint)
+	assert.NoError(t, err)
+	mockAuthCtx := mocks.AuthenticationContext{}
+	mockAuthCtx.On("GetBaseURL").Return(baseURL)
+	mockAuthCtx.On("GetMetadataURL").Return(metadataPath)
+	handler := GetMetadataEndpointRedirectHandler(ctx, &mockAuthCtx)
+	req, err := http.NewRequest("GET", "/xyz", nil)
+	assert.NoError(t, err)
+	w := httptest.NewRecorder()
+	handler(w, req)
+	assert.Equal(t, 303, w.Code)
```

Comment:
nit: use the named constant instead? https://golang.org/pkg/net/http/#pkg-constants